### PR TITLE
Update Kubernetes to v1.19.15

### PR DESCRIPTION
**Pull request by manuelbuil**
Signed-off-by: Manuel Buil <mbuil@suse.com>
#### Proposed Changes ####

Update Kubernetes

#### Types of Changes ####

Kubernetes

#### Verification ####

`kubectl version -o yaml`

#### Linked Issues ####

* https://github.com/k3s-io/k3s/issues/4006

#### User-Facing Change ####

```release-note
Update Kubernetes to v1.19.15
```

### CODE REVIEW

**Review by dereknola — APPROVED** [review]
Step 1 Approval

File: go.mod [hunk 1]
```diff
@@ -122,7 +122,7 @@ require (
 	k8s.io/component-base v0.19.14
```
Comment (brandond) [inline go.mod:122]: The submodules here that are at v0.19.14 should go to v0.19.15 for consistency's sake.

**Review by brandond — CHANGES_REQUESTED** [review]
LGTM to push tags, one nit on submodules.

**Comment by briandowns** [comment]
> Step 1 Approval

We should update the docs to have this as a required comment for this process.

File: go.mod (hunk 1 above)
Comment (manuelbuil) [reply go.mod]: danke schön! I think now it is correct

**Comment by manuelbuil** [comment]
> > Step 1 Approval
> 
> We should update the docs to have this as a required comment for this process.

tags pushed! https://github.com/k3s-io/kubernetes/releases/tag/v1.19.15-k3s1

File: go.mod [hunk 2]
```diff
@@ -114,15 +114,15 @@ require (
...
-	k8s.io/apiserver v0.19.14
+	k8s.io/api v0.19.15
+	k8s.io/apimachinery v0.19.15
+	k8s.io/apiserver v0.19.15
 	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
-	k8s.io/cloud-provider v0.19.14
-	k8s.io/component-base v0.19.14
+	k8s.io/cloud-provider v0.19.15
+	k8s.io/component-base v0.19.15
 	k8s.io/cri-api v0.20.1
```
Comment (brandond) [inline go.mod:123]: This one is part of the replaced k8s.io submodule set as well
```suggestion
	k8s.io/cri-api v0.19.15
```

**Review by brandond — CHANGES_REQUESTED** [review]

**Comment by brandond** [comment]
Upstream is building with golang 1.15.15, can you ensure that the golang version gets bumped as well?

File: go.mod (hunk 2 above)
Comment (manuelbuil) [reply go.mod:123]: If I change  that and run `go mod vendor` comes back to 0.20.1

Comment (brandond) [reply go.mod:123]: Nevermind, containerd's CRI stuff wants this pulled forward so it's fine.

**Review by brandond — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

**Pull request merged**